lib/code_corps/github/event/pull_request/task_syncer.ex

```diff
@@ -0,0 +1,65 @@
+defmodule CodeCorps.GitHub.Event.PullRequest.TaskSyncer do
+  alias CodeCorps.{
+    GithubPullRequest,
+    GithubRepo,
+    GitHub.Event.Common.ResultAggregator,
+    GitHub.Event.PullRequest.ChangesetBuilder,
+    ProjectGithubRepo,
+    Task,
+    User,
+    Repo
+  }
+
+  alias Ecto.Changeset
+
+  @type outcome :: {:ok, list(Task.t)} |
+                   {:error, {list(Task.t), list(Changeset.t)}}
+
+  @doc """
+  When provided a `CodeCorps.GithubPullRequest`, a `CodeCorps.User` and a
+  GitHub API payload, for each `CodeCorps.Project` associated to that
+  `CodeCorps.GithubRepo` via a `CodeCorps.ProjectGithubRepo`, it
+  creates or updates a `CodeCorps.Task`.
+  """
+  @spec sync_all(GithubPullRequest.t, User.t, map) :: {:ok, list(Task.t)}
+  def sync_all(%GithubPullRequest{} = github_pull_request, %User{} = user, %{} = payload) do
+
+    %GithubPullRequest{
+      github_repo: %GithubRepo{project_github_repos: project_github_repos}
+    } = github_pull_request |> Repo.preload(github_repo: :project_github_repos)
+
+    project_github_repos
+    |> Enum.map(&sync(github_pull_request, &1, user, payload))
+    |> ResultAggregator.aggregate
+  end
+
+  @spec sync(GithubPullRequest.t, ProjectGithubRepo.t, User.t, map) :: {:ok, ProjectGithubRepo.t} | {:error, Changeset.t}
+  defp sync(%GithubPullRequest{} = github_pull_request, %ProjectGithubRepo{} = project_github_repo, %User{} = user, %{} = payload) do
+    project_github_repo
+    |> find_or_init_task(github_pull_request)
+    |> ChangesetBuilder.build_changeset(payload, github_pull_request, project_github_repo, user)
+    |> commit()
+  end
+
+  @spec find_or_init_task(ProjectGithubRepo.t, GithubPullRequest.t) :: Task.t
+  defp find_or_init_task(
+    %ProjectGithubRepo{project_id: project_id, github_repo_id: github_repo_id},
+    %GithubPullRequest{id: github_pull_request_id}
+  ) do
```

Comment:
This doesn't deal with the key part we need here - the parsing of the PR body to inferr which tasks are to be linked.

Really, it's the core logic of that that we need to work out, I think.

- parse body for closing keywords, extract github ids
- if they exist, find github issues with those ids, associate with tasks belonging to those issues, dissasociate form tasks belonging to other github issues
- if none are found, disassociate from associated tasks which do belong to github issues

The special case is when we get a pr created which does not close any issues. In that case, my suggestion is we
- create a codecorps task, which is linked to the pr, but not to a github iusse
- we leave that task associated from now on, regardless of new associations being added/removed

My philosophy is, that one was created on CC and can potentially hold information that will be lost, if we disassociate. Maybe at some point in the future, we allow "converting the PR to a github issue", which would associate that task with a github issue as well. At that point, it can behave like a normal task, but until that point, we shouldn't assume and try to avoid any automatons with it. If the PR has been linked to an issue since, let the user close the original task we've created if they want to.